### DISCUSSION TITLE
Issue with cfg$title when running output.R

### DISCUSSION THREAD

**Discussion by Panda1i** [magpiemodel/magpie#607] [Help]
Hello,

I've been working with a script named `output.R` and encountered an error that I'm having trouble resolving. The error message is as follows:

Error in cfg$title : $ operator is invalid for atomic vectors

This error occurs when I attempt to run the script of validation in `output.R`, and it seems to be related to the configuration list `cfg` where `cfg$title` is used.

I have checked the code thoroughly and ensured that cfg is indeed a list and that cfg$title is being assigned a string value. Despite this, the error persists.

Could you please provide some insight into what might be causing this issue? Is there a mistake I made that could lead to this error?I am using R version 4.3.1 on a Windows system.

Any help or guidance would be greatly appreciated. Thank you for your time and assistance.

Best regards,
Panda1i

**Comment by tscheypidi** [comment]
Hi Panda,

could you provide some more information? Can you perhaps share the config you are using and the output script you are using? Alternatively, you could add a `print(cfg)` statement the line before the error happens or run the script with `options(error=recover)` to be able to investigate what the state of the cfg list is when the error occurs.

Best, Jan

**Comment by Panda1i** [comment]
Hi Jan,

Thank you for your prompt response. I have attached my [config and output script.zip](https://github.com/magpiemodel/magpie/files/13296674/config.and.output.script.zip) files for your review. Additionally, I have downloaded the latest version of the MAgPIE model from GitHub, and copied the output files from the provided model runs from Mishra(https://doi.org/10.5194/gmd-14-6467-2021), which was provided by the tutorial, into the new MAgPIE model directory. And I attempted to rerun output.R. Unfortunately, I encountered the same error: Error in cfg$title : $ operator is invalid for atomic vectors.

As a newcomer to MAgPIE, I have not yet run my own models and am relying on the output files from the MAgPIE workshop tutorials to practice. I'm hoping you might be able to shed some light on what might be going wrong.

Thank you for your time and assistance.

Best regards,
Panda1i

**Reply by tscheypidi** [reply]
sorry....I am still not able to follow. Can you describe again the steps you did? You downloaded the model and then ran `Rscript output.R`?

**Reply by Panda1i** [reply]
Apologies for any confusion in my previous message. Let me clarify the steps I've taken and the issue I'm encountering:

I have not yet conducted my own MAgPIE model simulation. Instead, I used the output files provided in the tutorial available at [MAgPIE Tutorial: Regional Output Analysis](https://magpiemodel.github.io/tutorials/t08-regionaloutputanalysis) to familiarize myself with the process.

I followed these steps:

1. I downloaded the output files from the tutorial link.
2. I placed these files into the output directory of the MAgPIE model.
3. I executed the output.R script, selecting the "all-validation-Direct execution" option.
4. At this point, I encountered the error message.
5. To ensure that the issue was not with the model setup, I downloaded a fresh copy of the MAgPIE model and repeated the process by copying the tutorial's output files into the new model's output folder.
6. Running the output.R script again resulted in the same error.

I have not made any modifications to the output.R and default.cfg files, this is why I'm puzzled about the source of the error. I'm beginning to wonder if the issue could be related to the fact that I haven't run my own simulations yet.

I'm very thankful for the time you're investing to help me resolve this problem.
